src/tls.rs

```diff
@@ -17,7 +17,7 @@ pub(crate) struct SwitchableConn<T: Read + Write>(pub(crate) Option<EitherConn<T
 
 pub(crate) enum EitherConn<T: Read + Write> {
     Plain(T),
-    Tls(rustls::StreamOwned<ServerConnection, PrependedReader<T>>),
+    Tls(Box<rustls::StreamOwned<ServerConnection, PrependedReader<T>>>),
```

Comment:
Why does this have to be `Box`ed now?

Reply:
also is clippy lint

<img width="1017" alt="Screen Shot 2022-12-10 at 18 39 45" src="https://user-images.githubusercontent.com/5784607/206851017-fb9a3b8c-bde1-4399-9a19-29e3a0737847.png">

Reply:
Ah, interesting. I'm torn on this one, but it may be a reasonable way to go if we expect that _many_ connections are going to be unencrypted. Given this is `pub(crate)` only, it's also something we can revert if we find that it has a detrimental impact, so I'm fine going with the lint's advice for now.